test_unstructured_ingest/dest/pinecone.sh

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env bash
+
+set -e
+
+DEST_PATH=$(dirname "$(realpath "$0")")
+SCRIPT_DIR=$(dirname "$DEST_PATH")
+cd "$SCRIPT_DIR"/.. || exit 1
+OUTPUT_FOLDER_NAME=s3-pinecone-dest
+OUTPUT_DIR=$SCRIPT_DIR/structured-output/$OUTPUT_FOLDER_NAME
+WORK_DIR=$SCRIPT_DIR/workdir/$OUTPUT_FOLDER_NAME
+max_processes=${MAX_PROCESSES:=$(python3 -c "import os; print(os.cpu_count())")}
+writer_processes=$(( (max_processes - 1) > 1 ? (max_processes - 1) : 2 ))
+
+if [ -z "$PINECONE_API_KEY" ]; then
+   echo "Skipping Pinecone ingest test because PINECONE_API_KEY env var is not set."
+   exit 0
+fi
+
+if [ -z "$PINECONE_INDEX" ]; then
+  # Reason of using RANDOM rather than uuidgen here is that index names generated with uuidgen
+  # were not compatible, giving a 400 response probably due to the length of or character variety in
+  # the index name.
+  RANDOM_SUFFIX=$((RANDOM % 100000 + 1))
+  PINECONE_INDEX="ingest-test-$RANDOM_SUFFIX"
+fi
+
+if [ -z "$PINECONE_ENVIRONMENT" ]; then
+  PINECONE_ENVIRONMENT="us-east1-gcp"
+fi
```

Comment:
You can set these using defaults [reference](https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html):
```suggestion
PINECONE_ENVIRONMENT=${PINECONE_ENVIRONMENT:-"us-east1-gcp"}
```

Reply:
Oh neat. The walrus (and 1-tooth family) lives in shell as well. I will definitely be using that)

Reply:
Added with [walrus syntax := instead of if [-z $...] for default parameters](https://github.com/Unstructured-IO/unstructured/pull/1774/commits/127837965117da83c7e5c17c6628c1990a051bd2)